functions_v1/src/index.js

```diff
@@ -114,39 +133,37 @@ const gotConfig = {
     }
 }
 
+api.use(compression())
+api.use(cors())
+
+api.get([
+    "/api/v2/"
+], (req, res) => {
+    fetchAndReply(req, res)
+})
+
 api.get([
-    "/api/v2/",
     "/api/v2/:endpoint/:id/",
     "/api/v2/:endpoint/:id/:extra/"
 ], (req, res) => {
-    got(targetUrlForPath(req.path), gotConfig)
-    .json()
-    .then(json => {
-        res.set('Cache-Control', `public, max-age=${successTtl}, s-maxage=${successTtl}`)
-        res.send(json)
-    })
-    .catch(reason => {
-        handleErrors(reason, req, res)
-    })
+    if (req.params.extra === undefined || req.params.extra === 'encounters') {
+        if (endpoints.includes(req.params.endpoint) && req.params.id.match(resources_r)) {
+            fetchAndReply(req, res)
+        } else {
+            res.sendStatus(400)
```

Comment:
404 is probably more correct than 400 here

Reply:
I wanted to return a `bad request` code. So I can differentiate about legit requests and broken requests in the GCP console. 

So that if someone queries `/pokemon/notexistent` get a 404
and if `/pokemon/notexist"ent` a 400